src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -404,6 +409,64 @@ void FancyZones::MoveWindowIntoZone(HWND window, winrt::com_ptr<IZoneWindow> zon
     }
 }
 
+int RectWidth(const RECT& rect)
+{
+    return rect.right - rect.left;
+}
+
+int RectHeight(const RECT& rect)
+{
+    return rect.bottom - rect.top;
+}
+
+RECT FitOnScreen(const RECT& windowRect, const RECT& originMonitorRect, const RECT& destMonitorRect)
+{
+    // New window position on active monitor. If window fits the screen, this will be final position.
+    int left = destMonitorRect.left + (windowRect.left - originMonitorRect.left);
+    int top = destMonitorRect.top + (windowRect.top - originMonitorRect.top);
+    int W = RectWidth(windowRect);
+    int H = RectHeight(windowRect);
+
+    if ((left < destMonitorRect.left) || (left + W > destMonitorRect.right))
+    {
+        // Set left window border to left border of screen (add padding). Resize window width if needed.
+        left = destMonitorRect.left + LEFT_TOP_PADDING;
+        W = min(W, RectWidth(destMonitorRect) - LEFT_TOP_PADDING);
+    }
+    if ((top < destMonitorRect.top) || (top + H > destMonitorRect.bottom))
+    {
+        // Set top window border to top border of screen (add padding). Resize window height if needed.
+        top = destMonitorRect.top + LEFT_TOP_PADDING;
+        H = min(H, RectHeight(destMonitorRect) - LEFT_TOP_PADDING);
+    }
+
+    return { left, top, left + W, top + H };
+}
+
+void OpenWindowOnActiveMonitor(HWND window, HMONITOR monitor) noexcept
+{
+    // By default windows opens new window on primary monitor.

```

Comment:
nit: typo 'windows' -> 'Windows` to make it clear

Reply:
Modified.